x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/processes/processes.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useCallback } from 'react';
+import { debounce } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiSearchBar,
+  EuiSpacer,
+  EuiEmptyPrompt,
+  EuiButton,
+  EuiText,
+  EuiIconTip,
+  Query,
+} from '@elastic/eui';
+import type { InventoryItemType } from '../../../../../../../common/inventory_models/types';
+import { getFieldByType } from '../../../../../../../common/inventory_models';
+import { parseSearchString } from '../../../../inventory_view/components/node_details/tabs/processes/parse_search_string';
+import { ProcessesTable } from '../../../../inventory_view/components/node_details/tabs/processes/processes_table';
+import { STATE_NAMES } from '../../../../inventory_view/components/node_details/tabs/processes/states';
+import { SummaryTable } from '../../../../inventory_view/components/node_details/tabs/processes/summary_table';
+import { TabContent } from '../../../../inventory_view/components/node_details/tabs/shared';
+import {
+  SortBy,
+  useProcessList,
+  ProcessListContextProvider,
+} from '../../../../inventory_view/hooks/use_process_list';
+import type { HostNodeRow } from '../../../hooks/use_hosts_table';
+import { useHostFlyoutOpen } from '../../../hooks/use_host_flyout_open_url_state';
+
+interface ProcessesProps {
+  node: HostNodeRow;
+  nodeType: InventoryItemType;
+  currentTime: number;
+}
+
+export const Processes = ({ currentTime, node, nodeType }: ProcessesProps) => {
+  const [hostFlyoutOpen, setHostFlyoutOpen] = useHostFlyoutOpen();
+  const [searchBarState, setSearchBarState] = useState<Query>(
```

Comment:
**nit:** use the lazy init for this state to prevent parsing the `hostFlyoutOpen.searchFilter` unnecessarily after the first render.

Reply:
Thanks for the suggestion! done 👍